big_tests/tests/sm_SUITE.erl

```diff
@@ -177,6 +181,48 @@ server_returns_failed(Config, ConnActions) ->
     escalus:assert(is_sm_failed, [<<"unexpected-request">>],
                    escalus_connection:get_stanza(Alice, enable_sm_failed)).
 
+client_enables_sm_twice_fails_with_correct_error_stanza(Config) ->
+    AliceSpec = escalus_fresh:create_fresh_user(Config, alice),
+    Steps = connection_steps_to_session(),
+    {ok, Alice, Features} = escalus_connection:start(AliceSpec, Steps),
+    escalus_session:stream_management(Alice, Features),
+    escalus_connection:send(Alice, escalus_stanza:enable_sm()),
+    escalus:assert(is_sm_failed, [<<"unexpected-request">>],
+                   escalus_connection:get_stanza(Alice, enable_sm_failed)),
+    escalus:assert(is_stream_end,
+                   escalus_connection:get_stanza(Alice, enable_sm_failed)),
+    wait_until_client_disconnects(Alice).
+
+session_resumed_then_old_session_is_closed_gracefully_with_correct_error_stanza(Config) ->
+    %% GIVEN USER WITH STREAM RESUMPTION ENABLED
+    {Alice, AliceSpec, SMID, SMH} =
+        get_stream_resumption_enabled_fresh_user_smid_and_h(Config, alice),
+    %% WHEN USER RESUMES SESSION FROM NEW CLIENT
+    Steps2 = connection_steps_to_stream_resumption(SMID, SMH),
+    {ok, Alice2, _} = escalus_connection:start(AliceSpec, Steps2),
+    process_initial_stanza(Alice2),
+    %% THEN: Old session is gracefully closed with the correct error stanza
+    escalus:assert(is_stream_error, [<<"conflict">>, <<>>],
+                   escalus_connection:get_stanza(Alice, close_old_stream)),
+    escalus:assert(is_stream_end,
+                   escalus_connection:get_stanza(Alice, close_old_stream)),
+    wait_until_client_disconnects(Alice),
+    true = escalus_connection:is_connected(Alice2),
```

Comment:
The best way to check if connection is still working is to send and receive stanza between two users.